src/nrniv/multisplit.cpp

```diff
@@ -886,11 +887,12 @@ bb_relation[j], rthost[j]);
 		for (j=0; j < n; ++j) {
 			if (rthost[j] == nrnmpi_myid) {
 				// sid to rank table
-				Int2IntTable* s2rt = new Int2IntTable(20);
-				int rank = 0; int mapsize = 0; int r;
+				Int2IntTable* s2rt = new Int2IntTable();
+				s2rt->reserve(20);
+				int rank = 0; int mapsize = 0;
 				for (k=0; k < nt; ++k) {
 					if (mark[k] == j && all_bb_relation[k] >= 2) {
-						if (!s2rt->find(allsid[k], r)) {
+						if (!s2rt->count(allsid[k])) {
 							(*s2rt)[allsid[k]] = rank++;
 						}
```

Comment:
```suggestion
						auto const result = s2rt->insert({allsid[k], rank});
						if (result.second) { rank++; }
```
avoids looking up twice